pennylane/estimator/resources_base.py

```diff
@@ -0,0 +1,407 @@
+              'Hadamard': 15,
+              'Z': 5,
+              'CNOT': 4
+
+
+        Specifically, users can add together two instances of resources using the :code:`+` and
+        :code:`&` operators. These represent combining the resources assuming the circuits were
+        executed in series or parallel respectively.
+
+        .. code-block:: pycon
+
+            >>> res_in_series = res1 + res2
+            >>> print(res_in_series)
+            --- Resources: ---
+             Total qubits: 9
+                algorithmic qubits: 4
+                allocated qubits: 5
+                     clean qubits: 2
+                     dirty qubits: 3
+             Total gates : 41
+              'Hadamard': 25,
+              'X': 5,
+              'CNOT': 6,
+              'Z': 5
+
+            >>> res_in_parallel = res1 & res2
+            >>> print(res_in_parallel)
+            --- Resources: ---
+             Total qubits: 12
+                algorithmic qubits: 7
+                allocated qubits: 5
+                     clean qubits: 2
+                     dirty qubits: 3
+             Total gates : 41
+              'Hadamard': 25,
+              'X': 5,
+              'CNOT': 6,
+              'Z': 5
+
+        Similarly, users can scale up the resources for an operator by some integer factor using
+        the :code:`*` and :code:`@` operators. These represent scaling the resources assuming the
+        circuits were executed in series or parallel respectively.
+
+        .. code-block:: pycon
+
+            >>> res_in_series = 5 * res1
+            >>> print(res_in_series)
+            --- Resources: ---
+             Total qubits: 10
+                algorithmic qubits: 3
+                allocated qubits: 7
+                     clean qubits: 2
+                     dirty qubits: 5
+             Total gates : 85
+              'Hadamard': 50,
+              'X': 25,
+              'CNOT': 10
+
+            >>> res_in_parallel = 5 @ res1
+            >>> print(res_in_parallel)
+            --- Resources: ---
+             Total qubits: 22
+                algorithmic qubits: 15
+                allocated qubits: 7
+                     clean qubits: 2
+                     dirty qubits: 5
+             Total gates : 85
+              'Hadamard': 50,
+              'X': 25,
+              'CNOT': 10
+
+    """
+
+    def __init__(self, wire_manager: WireResourceManager, gate_types: dict = None):
+        """Initialize the Resources class."""
+        gate_types = gate_types or {}
+
+        self.wire_manager = wire_manager
+        self.gate_types = (
+            gate_types
+            if (isinstance(gate_types, defaultdict) and isinstance(gate_types.default_factory, int))
+            else defaultdict(int, gate_types)
+        )
+
+    def __add__(self, other: Resources) -> Resources:
+        """Add two resources objects in series"""
+        assert isinstance(other, self.__class__)
+        return add_in_series(self, other)
```

Comment:
Why do we need both the dunder method and an `add_in_series` function? Why not just have the dunder method?